wire-ios-data-model/Source/Utilis/NSPredicate+BaseCompounds.swift

```diff
@@ -0,0 +1,35 @@
+////
+// Wire
+// Copyright (C) 2023 Wire Swiss GmbH
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see http://www.gnu.org/licenses/.
+//
+
+import Foundation
+
+public extension NSPredicate {
+
+    var inverse: NSPredicate {
+        return NSCompoundPredicate(notPredicateWithSubpredicate: self)
+    }
+
+    func and(_ other: NSPredicate) -> NSPredicate {
+        return NSCompoundPredicate(andPredicateWithSubpredicates: [other])
+    }
+
+    func or(_ other: NSPredicate) -> NSPredicate {
+        return NSCompoundPredicate(orPredicateWithSubpredicates: [other])
+    }
```

Comment:
Are these meant to include `self` in the returned compound predicate? It looks like it's only creating a compound predicate with `other`.

Reply:
fixed